api/envoy/extensions/filters/http/header_mutation/v3/header_mutation.proto

```diff
@@ -33,4 +34,17 @@ message HeaderMutationPerRoute {
 // always be applied first and then the per-route mutation rules, if both are specified.
 message HeaderMutation {
   Mutations mutations = 1;
+
+  // If per route configuration is configured at multiple levels (route, virtual host, and route table),
+  // all specified levels' mutations are applied. By default, header mutations are evaluated from least
```

Comment:
When I look at the companion comment for the similar field in route.proto, I'm actually super confused, it's actually a `bool` and I don't see how it can have those default semantics. What's going on?

Reply:
The order here actually is the reversed order of route's sequence
 - `route proto` is evaluated from `most to least` by default ([proto](https://github.com/envoyproxy/envoy/blob/57f23d3092956972f2c23f869280310c7a65ca33/api/envoy/config/route/v3/route.proto#L85) and [implementation](https://github.com/envoyproxy/envoy/blob/57f23d3092956972f2c23f869280310c7a65ca33/source/common/router/config_impl.cc#L171)). i.e., most_specific_wins = false

- `header_mutation` here is evaluated from `least to most` by default. i.e., most_specific_wins = true. 

The reason I chosen this way is because header_mutation's old behavior is [only most specific is used ](https://github.com/envoyproxy/envoy/blob/57f23d3092956972f2c23f869280310c7a65ca33/source/extensions/filters/http/header_mutation/header_mutation.cc#L37). So I was trying to match old behavior as default. If you think the consistency between route and header_mutation is more important/clearer, I am happy to change it.

Reply:
Or we can name this field as `least_specific_header_mutations_wins` to avoid confuse. But yes, I like the consistent behavior between this filter and router config, but we probably need to add new runtime flag to change that behavior. I wasn't that is worth or not. Maybe just use another field name to avoid this confuse.

Reply:
I think having this consistency is good for these knobs.

I don't understand: 

> route proto is evaluated from most to least by default ([proto](https://github.com/envoyproxy/envoy/blob/57f23d3092956972f2c23f869280310c7a65ca33/api/envoy/config/route/v3/route.proto#L85) and [implementation](https://github.com/envoyproxy/envoy/blob/57f23d3092956972f2c23f869280310c7a65ca33/source/common/router/config_impl.cc#L171)). i.e., most_specific_wins = false

How can the behavior of evaluating from most to least specific (with most specific winning presumably) correspond to most_specific_wins = false?

Reply:
The logic here is: later evaluation wins. [Impl reference](https://github.com/envoyproxy/envoy/blob/99ebb7c3b8350d50aa772e57705f3ffbf8772bdd/source/common/router/config_impl.cc#L821-L822). It is because the later evaluation can override the earlier ones (i.e., wins).

Sure I will update the code here to be consistent with router (presumably with the same name `most_specific_wins`, default false)

Reply:
Done. Updated both route.proto and head_mutation proto header. 
PTAL